### PR TITLE
feat: custom vesting module to check signer in allow list

### PR DESCRIPTION
**Pull request by trinitys7**
As mentioned in [rollapp-evm ](https://github.com/dymensionxyz/rollapp-evm/issues/169). We need to custom vesting module for creating a allow list that can create vest account.
The code is taken from [cosmos-sdk ](https://github.com/cosmos/cosmos-sdk/tree/v0.46.16/x/auth/vesting) with a few changes for `allow list`.

### CODE REVIEW

File: x/vesting/keeper/permissioned_decorator.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+// AnteHandle rejects vesting messages that signer does not have permissions
+// to create vesting account.
+func (pvd PermissionedVestingDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, next sdk.AnteHandler) (newCtx sdk.Context, err error) {
+	for _, msg := range tx.GetMsgs() {
+		typeURL := sdk.MsgTypeURL(msg)
+		for _, disabledTypeURL := range pvd.disabledMsgTypeURLs {
+			if typeURL == disabledTypeURL {
+				// Check if vesting tx signer is 1
+				if len(msg.GetSigners()) != 1 {
```
Comment (omritoptix) [inline x/vesting/keeper/permissioned_decorator.go:30]: why are we blocking multisigs?

**Review by omritoptix — COMMENTED** [review]
can we write genesis tests and ante handler test please?

File: x/vesting/keeper/permissioned_decorator.go (hunk 1 above)
Comment (trinitys7) [reply x/vesting/keeper/permissioned_decorator.go:30]: As I see, all vesting msgs on [cosmos-sdk](https://github.com/cosmos/cosmos-sdk/blob/d6d929843bbd331b885467475bcb3050788e30ca/x/auth/vesting/types/msgs.go#L76) or [evmos](https://github.com/dymensionxyz/evmos/blob/4ab8372bced4a8509de9abc4c0ad53b50e34b1b3/x/vesting/types/msg.go#L121) only contains 1 signer. WDYT

**Comment by mtsitrin** [comment]
@omritoptix isn't it bit redundent to have a dedicated module to keep the permission address?
the module itself doesn't have any logic, just provides the ante decorator, which can leave outside a module.

Can't we reuse the whitelisted address from other module? it would be the 3rd place we manage `whitelist` param

**Comment by mtsitrin** [comment]
we need to consider how to enforce it through `Authz` flow
* either run same verification on `Authz` flow
* or block vesting Msg through `Authz`

File: x/vesting/keeper/permissioned_decorator.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
+package keeper
+
+import (
+	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/dymensionxyz/dymension-rdk/x/vesting/types"
+)
+
+// PermissionedVestingDecorator prevents invalid msg types from being executed
+type PermissionedVestingDecorator struct {
```
Comment (mtsitrin) [inline x/vesting/keeper/permissioned_decorator.go:10]: not wired in app.go ante handler

**Comment by omritoptix** [comment]
> Can't we reuse the whitelisted address from other module? it would be the 3rd place we manage whitelist param

@mtsitrin yea thought about it initially but imo it's less than ideal for 2 reasons:
1. both modules which have currently the whitelisted addresses (i.e `hubgenesis` and `denom-metadata`) are temporary as we plan to automate this through ibc from the hub but the vesting whitelist is supposedly a long term solution.
2. it's not necessarily the same actors (i.e you may want to give whitelist vesting permissions to a wider crowd)

File: x/vesting/keeper/permissioned_decorator.go (hunk 1 above)
Comment (omritoptix) [reply x/vesting/keeper/permissioned_decorator.go:30]: ok we can start with that

File: x/vesting/keeper/permissioned_decorator.go (hunk 2 above)
Comment (trinitys7) [reply x/vesting/keeper/permissioned_decorator.go:10]: the dymension-rdk doesn't actually have antehandler. So I added it here and on rollapp-evm's app.go, we just need to call the Decorator from dymension-rdk

**Comment by trinitys7** [comment]
Close and decide to not have vesting module on rdk just for permissioned addresses

**Pull request closed without merging**